pkg/splunk/enterprise/names.go

```diff
@@ -124,6 +124,37 @@ const (
 	livenessProbeDefaultDelaySec = 300
 	livenessProbeTimeoutSec      = 30
 	livenessProbePeriodSec       = 30
+
+	// Number of ClusterMasterReplicas
+	numberOfClusterMasterReplicas = 1
+
+	// Number of Deployer replicas
+	numberOfDeployerReplicas = 1
+
+	// Number of Licensemaster replicas
+	numberOfLicenseMasterReplicas = 1
```

Comment:
Ideally, we should be able to use the CR replica count value? If the reason for this change is invalid value for the replica count, probably, it is high time for validating the same during the spec validation time, with a default and optional values.

Reply:
The LM, CM CRs don't have a replica field in the spec. The SHC CR doesn't have a replica count for deployer?

Reply:
@sgontla Are you suggesting introducing new `replica` fields in the CM, LM and SHC CR(for deployer)?

Reply:
As discussed, we will take care of this at a later time as the controller has logic to initialize default values only when the spec field is at-least an empty struct while creating the CR.

**Details**:

I have two LM's(one with an empty spec struct and the other with no spec struct declared at all):

```
apiVersion: enterprise.splunk.com/v3
kind: LicenseMaster
metadata:
  name: test-with-spec
  finalizers:
  - enterprise.splunk.com/delete-pvc
spec: {}
---
apiVersion: enterprise.splunk.com/v3
kind: LicenseMaster
metadata:
  name: test-no-spec
  finalizers:
  - enterprise.splunk.com/delete-pvc
```

For the `test-with-spec` LM I see that the replica count is 1(which is set a default value with kubebuilder) while the `test-no-spec` has replicas 0.